### PR TITLE
SFD-3376 Increasing code coverage

### PR DESCRIPTION
**Pull request by ArifNhs**
SFD-3376 Increasing code coverage

### CODE REVIEW

File: application/src/test/java/uk/nhs/digital/uec/api/controller/PostcodeMappingControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  @Test
+  public void testName() throws InvalidPostcodeException {
+    when(postcodeService.getByName(serviceName)).thenReturn(Arrays.asList(postcodeMapping));
+    ResponseEntity<?> response = postcodeMappingController.getPostcodeMapping(null, serviceName);
+
+    assertEquals(HttpStatus.OK, response.getStatusCode());
+  }
+}
```
Comment (daonhs) [inline application/src/test/java/uk/nhs/digital/uec/api/controller/PostcodeMappingControllerTest.java:64]: Perhaps a scenario for when an InvalidPostCodeException is thrown

File: application/src/test/java/uk/nhs/digital/uec/api/service/PostcodeMappingServiceTest.java [hunk 2]
```diff
@@ -24,73 +22,68 @@
...
+  public void testEastingAndNorthing() throws InvalidPostcodeException {
+    when(postcodeMappingRepository.findByName(serviceName)).thenReturn(postcodeMappingOptList);
+    List<PostcodeMapping> findByPostCodeIn = postcodeMappingService.getByName(serviceName);
+    PostcodeMapping postCodeMapping = findByPostCodeIn.stream().findAny().get();
+    int easting = postCodeMapping.getEasting();
+    int northing = postCodeMapping.getNorthing();
+    assertEquals(12345, easting);
+    assertEquals(360, northing);
+  }
 }
```
Comment (daonhs) [inline application/src/test/java/uk/nhs/digital/uec/api/service/PostcodeMappingServiceTest.java:89]: Need to test an error scenario

**Review by daonhs — COMMENTED** [review]
Just 2 tests for error scenarios and should be okay

File: application/src/test/java/uk/nhs/digital/uec/api/controller/PostcodeMappingControllerTest.java (hunk 1 above)
Comment (ArifNhs) [reply application/src/test/java/uk/nhs/digital/uec/api/controller/PostcodeMappingControllerTest.java:64]: In controller test the service layer is mocked, so all the negative scenarios is included in Util test class (PostcodeUtils)

**Review by daonhs — APPROVED** [review]

File: application/src/test/java/uk/nhs/digital/uec/api/service/PostcodeMappingServiceTest.java (hunk 2 above)
Comment (ArifNhs) [reply application/src/test/java/uk/nhs/digital/uec/api/service/PostcodeMappingServiceTest.java:89]: The service layer is mocked, so all the negative scenarios is included in Util test class (PostcodeUtils)

**Review by jope1 — APPROVED** [review]

**Pull request merged**